src/pynwb/ndx_patterned_ogen/patterned_ogen.py

```diff
@@ -480,27 +525,72 @@ def __init__(self, **kwargs):
         super().__init__(**kwargs)
         for key, val in args_to_set.items():
             setattr(self, key, val)
+        columns = popargs("columns", kwargs)
+        if columns is not None:
+            colset = {c.name: c for c in columns}
+            # First check: power_per_rois and power must not be defined in the same time
+            if "power_per_rois" in colset.keys() and "power" in colset.keys():
+                raise ValueError(
+                    "Both 'power' and 'power_per_rois' have been defined. Only one of them must be defined"
+                )
+
+            # Second check: all elements in "power", "frequency", "pulse_width" must be scalars
+            for column_to_check in ["power", "frequency", "pulse_width"]:
+                if column_to_check in colset.keys():
+                    self.check_if_argument_is_list(colset=colset, field_name=column_to_check)
+
+            # Third check: all elements in "power_per_rois", "frequency_per_rois", "pulse_width_per_rois" columns
+            # must be the same length as the respective targets
+            for column_to_check in ["power_per_rois", "frequency_per_rois", "pulse_width_per_rois"]:
+                if column_to_check in colset.keys():
+                    self.check_length_rois_properties(colset=colset, field_name=column_to_check)
 
     @docval(
         {"name": "start_time", "doc": "Start time of stimulation, in seconds.", "type": float},
         {"name": "stop_time", "doc": "Stop time of stimulation, in seconds.", "type": float},
         {
             "name": "power",
-            "doc": "Power (in Watts) applied to each target during patterned photostimulation.",
+            "doc": "Power (in Watts) defined as a scalar. All rois in target receive the same photostimulation power.",
             "type": (int, float, Iterable),
-            "default": 0.0,
+            "default": None,
         },
         {
             "name": "frequency",
-            "doc": "Frequency of stimulation if the stimulus delivered is pulsed (in Hz).",
+            "doc": (
+                "Frequency (in Hz) defined as a scalar. All rois in target receive the photostimulation at the same"
+                " frequency."
+            ),
             "type": (int, float, Iterable),
-            "default": 0.0,
+            "default": None,
         },
         {
             "name": "pulse_width",
-            "doc": "Pulse width of stimulation if the stimulus delivered is pulsed, in seconds/phase.",
+            "doc": (
+                "Pulse Width (in sec/phase) defined as a scalar. All rois in target receive the photostimulation with"
+                " the same pulse width."
+            ),
+            "type": (int, float, Iterable),
+            "default": None,
+        },
+        {
+            "name": "power_per_rois",
```

Comment:
A fairly minor point, but I read the nomenclature as follows...

A field named `rois` (plural) is a list of many individual `roi`

A field named `power_per_rois` to be a list of lists of many individual `roi`

So it would be more precise here to say `power_per_roi` as a list of power matching one-to-one for many individual `roi`

What do you think?

Reply:
Actually, I noticed too, but forgot to change them. I will change to ..._per_roi